ocfweb/docs/docs/services/hpc.md

```diff
@@ -0,0 +1,173 @@
+We also have a #hpc_users channel on [slack][fco] and [[irc|doc contact/irc]]
+where you can ask questions and talk to us about anything HPC.
+
+## The Cluster
+
+Users are currently able to run jobs on our main HPC server, which has 4
+1080Ti's and 256GB of DDR4-2400 RAM, which was funded by generous grants from
+the Student Tech Fund.
+
+## Slurm
+
+We currently use [Slurm][slurm] as our workload manager for the cluster. We
+will soon post technical documentation about our Slurm configuration, but briefly,
+Slurm is a free and open-source job scheduler which helps distribute jobs from
+all users evenly among HPC computers, referred to as nodes. Berkeley Research
+Computing (BRC) has some useful documentation for using Slurm [here][brc_slurm],
+just keep in mind that there are some differences between their configuration and
+ours. To put it simply, all of your programs will be run through Slurm. To use
+Slurm there are several commands that will be helpful:
+
+* `srun`: Used to submit jobs.
+* `scontrol`: Used to view and edit the Slurm configuration.
+* `squeue`: Used to view running and queued jobs.
+* `scancel`: Used to cancel jobs.
+* `sinfo`: Used to view status of compute nodes in a cluster.
+
+## Dependencies
+For managing application dependencies, you currently have two options:
+
+### Virtual Environments
+
+First you can use a [virtual environment][venv] if you are using Python
+packages. To create a virtual environment navigate to your home directory
+and run the following commands:
+
+```
+virtualenv -p python3 venv
+. venv/bin/activate
+```
+
+This will allow you to `pip install` any Python packages that the OCF does not
+already have for your program.
+
+### Singularity
+
+For those who need access to non-Python dependencies or have already integrated
+their program into Docker, the second option is to use Singularity containers.
+[Singularity][singularity] is a containerization platform developed at Lawrence
+Berkeley National Laboratory that is designed specifically for HPC environments.
+To read more about the benefits of Singularity you can look
+[here][singularity_article]. We suggest a particular workflow, which will help
+simplify deploying your program on our infrastructure.
+
+#### Installing
+
+We recommend that you do your development on our HPC infrastructure, but you
+can also develop on your own machine if you would like. If you are running
+Linux on your system, you can install Singularity from the official apt repos:
+
+```
+sudo apt install singularity-container
+```
+
+If you do not have an apt-based Linux distribution, installation instructions
+can be found [here][linux_install]. Otherwise, if you are running Mac you can
+look [here][mac_install], or Windows [here][win_install].
+
+#### Building Your Container
+
+```
+singularity build [--sandbox] ./my_container docker://ubuntu
+```
+This will create a Singularity container named `my_container`. If you are still
```

Comment:
I would suggest better organizing the logic in this paragraph wrt the workflows for development on the local computer vs development on our server. Perhaps something along the lines of:

> You can either run this command on your own computer or our server. If you run it on our infrastructure you will not be able to install non-pip packages inside your container because blah blah blah.
>
> If you run this command on your own computer, you will be able to install packages inside your container. But you'll want to prepend the command with `sudo` and specify `--sandbox` as well. Then you can copy the container over to our infrastructure by &lt;insert method here&gt;. Blah blah blah
>
> The `docker://ubuntu` argument blah blah blah